lib/vintage_net_wizard.ex

```diff
@@ -14,12 +16,17 @@ defmodule VintageNetWizard do
   Options:
 
     - `:ssl` - A Keyword list of `:ssl.tls_server_options`
+    - `:on_exit` - `{module, function, args}` tuple specifying
+    callback to perform after stopping the server.
 
   See `Plug.SSL.configure/1` for more information about the
   SSL options.
   """
   @spec run_wizard([Endpoint.opt()]) :: :ok | {:error, String.t()}
   def run_wizard(opts \\ []) do
+    Keyword.take(opts, [:on_exit])
```

Comment:
How about removing `Registry` here and handling `:on_exit` completely in `Endpoint`? Also, it looks like the `:on_exit` value could be saved as is. No processes are involved, so `Registry` isn't needed. Since there's only ever one callback, I don't think any data structure is needed.